pyam/utils.py

```diff
@@ -344,29 +350,51 @@ def _format_data_to_series(df, index):
 def format_data(df, index, **kwargs):
     """Convert a pandas.Dataframe or pandas.Series to the required format"""
 
-    if isinstance(df, pd.Series):
-        if not df.name:
-            df = df.rename("value")
-        df = df.reset_index()
-    elif not list(df.index.names) == [None]:
-        # reset the index if meaningful entries are included there
-        df = df.reset_index()
+    if set(df.index.names) >= set(index) | set(REQUIRED_COLS) and not kwargs:
+        # Let's try to cut corners here, it's our fast-path
+        time_col, extra_cols, data_cols = _intuit_column_groups(df, index=index)
+
+        if isinstance(df, pd.DataFrame):
+            extra_cols_not_in_index = [c for c in extra_cols if c in df.columns]
+            if extra_cols_not_in_index:
+                df = df.set_index(extra_cols_not_in_index, append=True)
 
-    df = _convert_r_columns(df)
+            if data_cols:
+                df = df[data_cols]
```

Comment:
slight preference for 
```suggestion
                df = (
                    df[data_cols]
                    .rename_axis(columns=time_col)
                    .stack()
                    .rename("value")
                    )
```

Reply:
Sure, sure, but I'll not commit your suggestion online, since stickler won't like the hanging brace.